source_modelling/sources.py

```diff
@@ -569,6 +647,16 @@ def __post_init__(self) -> None:
             for i in reversed(list(nx.topological_sort(points_into_graph)))
         ]
 
+    @property
+    def dip(self) -> float:
```

Comment:
If this can vary per plane, not sure if just returning the first one is the correct approach? Maybe the average (and rename to `avg_dip`)?

Reply:
Dip actually cannot vary per plane of a Fault. Supposing you did have two planes with a different dip, then they would not properly meet at an edge. It'd look like the picture below.

![image](https://github.com/user-attachments/assets/80436364-5cf4-420c-8911-3744ab1b58f1)

Reply:
Accordingly, I have added that consistency check back into the sources module.